### PR TITLE
A lot of fixes

### LINKED ISSUE #1921: [1.10] Several Issues Farmer, Cook, Baker
### Minecolonies version

version minecolonies-universal-1.10.2-0.9.6297

### Expected behavior
Issues:
 1. Farmer: Fields automaticly assined to one farmer
 2. Cook: Restaurants get deleted after Deconstruction
 3. Baker: Consumes resources while producing

### Actual behaviour

1. Farmer: 
 fields do not get assigned or assigned to multible Farmers at the same time
 if you assign manually not all field get recognized and assignment gets deleted every Server restart

issue_farmer

2. Cook: 
 For each Deconstructed Restaurant "Unknown" Cook keeps calling for food 

issue_restaurant

3. Baker: 
    not cosumes Weat but infinitly producing Bread

### Steps to reproduce the problem
 1. Farmer: 
    Step 1: Build Farmer
    Step 2: Build Farm
 2. Cook:
    Step 1: Build Restaurant
    Step 2: Wait for Cook to get Assigned
    Step 3: Deconstruct Restaurant
 3. Baker:
    Step 1: Build Bakery
    Step 2: Wait for Baker to get Assigned
    Step 3: Put 2 Stacks of Weat into Chest
    Step 4: Wait for Inifinite Bread

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #1921 

# Changes proposed in this pull request:
- Should fix several problems with the baker
- Should fix problems with workers not taking items or throwing exceptions
- Should fix problems with workers using their furnaces as containers
- Should fix the synch problems with the scarecrows

Review please

### CODE REVIEW

**Review by marchermans — DISMISSED** [review]

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 1]
```diff
@@ -33,7 +33,9 @@
 import com.minecolonies.coremod.util.StructureWrapper;
 import io.netty.buffer.ByteBuf;
 import net.minecraft.block.Block;
+import net.minecraft.block.BlockChest;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MINOR Remove this unused import 'net.minecraft.block.BlockChest'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 2]
```diff
@@ -33,7 +33,9 @@
...
 import net.minecraft.block.BlockContainer;
+import net.minecraft.block.BlockFurnace;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MINOR Remove this unused import 'net.minecraft.block.BlockFurnace'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 3]
```diff
@@ -11,6 +11,7 @@
 import com.minecolonies.api.util.*;
 import com.minecolonies.api.util.constant.IToolType;
 import com.minecolonies.api.util.constant.ToolType;
+import com.minecolonies.api.util.constant.TypeConstants;
 import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
 import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
 import com.minecolonies.coremod.colony.buildings.BuildingCook;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: MINOR Remove this unused import 'com.minecolonies.coremod.colony.buildings.BuildingCook'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 4]
```diff
@@ -11,6 +11,7 @@
 import com.minecolonies.api.util.*;
 import com.minecolonies.api.util.constant.IToolType;
 import com.minecolonies.api.util.constant.ToolType;
+import com.minecolonies.api.util.constant.TypeConstants;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.TypeConstants'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 5]
```diff
@@ -11,6 +11,7 @@
...
 import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: MINOR Remove this unused import 'com.minecolonies.coremod.colony.buildings.AbstractBuilding'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/EntityAIWorkBaker.java [hunk 6]
```diff
@@ -24,8 +24,10 @@
 import org.jetbrains.annotations.Nullable;
 
 import java.util.ArrayList;
+import java.util.Arrays;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/EntityAIWorkBaker.java]: MINOR Remove this unused import 'java.util.Arrays'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/EntityAIWorkBaker.java [hunk 7]
```diff
@@ -24,8 +24,10 @@
...
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/EntityAIWorkBaker.java]: MINOR Remove this unused import 'java.util.stream.Collectors'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 106 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 76 major
* ![MINOR][MINOR] 23 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1440](https://github.com/Minecolonies/minecolonies/blob/d9765148e18af0b8c8b7a1e518039dfc3ee1321d/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1440): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1469](https://github.com/Minecolonies/minecolonies/blob/d9765148e18af0b8c8b7a1e518039dfc3ee1321d/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1469): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1480](https://github.com/Minecolonies/minecolonies/blob/d9765148e18af0b8c8b7a1e518039dfc3ee1321d/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1480): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1586](https://github.com/Minecolonies/minecolonies/blob/d9765148e18af0b8c8b7a1e518039dfc3ee1321d/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1586): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [CombinedItemHandler.java#L276](https://github.com/Minecolonies/minecolonies/blob/d9765148e18af0b8c8b7a1e518039dfc3ee1321d/src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java#L276): Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [CombinedItemHandler.java#L278](https://github.com/Minecolonies/minecolonies/blob/d9765148e18af0b8c8b7a1e518039dfc3ee1321d/src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java#L278): Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/d9765148e18af0b8c8b7a1e518039dfc3ee1321d/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/d9765148e18af0b8c8b7a1e518039dfc3ee1321d/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/d9765148e18af0b8c8b7a1e518039dfc3ee1321d/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/d9765148e18af0b8c8b7a1e518039dfc3ee1321d/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 98 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 76 major
* ![MINOR][MINOR] 15 minor
* ![INFO][INFO] 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1438](https://github.com/Minecolonies/minecolonies/blob/a53835844dcbea3268e7887757840780ad8503e8/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1438): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1467](https://github.com/Minecolonies/minecolonies/blob/a53835844dcbea3268e7887757840780ad8503e8/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1467): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1478](https://github.com/Minecolonies/minecolonies/blob/a53835844dcbea3268e7887757840780ad8503e8/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1478): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1584](https://github.com/Minecolonies/minecolonies/blob/a53835844dcbea3268e7887757840780ad8503e8/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1584): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [CombinedItemHandler.java#L276](https://github.com/Minecolonies/minecolonies/blob/a53835844dcbea3268e7887757840780ad8503e8/src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java#L276): Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [CombinedItemHandler.java#L278](https://github.com/Minecolonies/minecolonies/blob/a53835844dcbea3268e7887757840780ad8503e8/src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java#L278): Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/a53835844dcbea3268e7887757840780ad8503e8/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/a53835844dcbea3268e7887757840780ad8503e8/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/a53835844dcbea3268e7887757840780ad8503e8/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/a53835844dcbea3268e7887757840780ad8503e8/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**